src/ui/wxWidgets/TreeCtrl.h

```diff
@@ -51,6 +52,44 @@ class TreeCtrl;
 
 typedef std::map<pws_os::CUUID, wxTreeItemId, std::less<pws_os::CUUID> > UUIDTIMapT;
 
+class TreeCtrl;
+
+class TreeScrollTimer: public wxTimer
+{
+public:
+    // start scrolling 1/4 second (if the mouse hasn't been clicked)
+  enum { DELAY = 250 };
+    
+  TreeScrollTimer();
+  
+  void setOwner(TreeCtrl *owner) { m_owner = owner; };
+
+  virtual void Notify();
+
+private:
+  TreeCtrl *m_owner;
+
+  wxDECLARE_NO_COPY_CLASS(TreeScrollTimer);
+};
+
+class TreeCollapseTimer: public wxTimer
```

Comment:
Now that we have different uses (YubiMixin, TreeCtrl) for timers, it would be a good opportunity to introduce a generic timer implementation at wxUtilities.

Reply:
The generic timer is wxTimer, as the specific one, with dedicated time and handler is the generate, as the given reference.

Reply:
By generic I mean a timer implementation that better suits the needs of the project so that a new class implementation does not have to be created for each new timer, that follows the same pattern and which only encapsulates the timeout value and a handler. Both information, the timeout value and the callback function, could be specified when setting up the timer. Such a timer could also offer a parameterless start function. If it already knows the timeout value to be used there is no need to pass the already known value to its own function again. I just think about preventing code duplication. - This is just an idea.

Reply:
Please provide a proposal class, I'm not good in implementing virtual classes/functions.

Reply:
The idea is to encapsulate the handler function as well, perhaps by replacing the specific receiver of type wxTreeCtrl with an interface/abstract class/lambda. The timer implementation could provide a handler registration function which is then called within the notification function, as is already the case in the current version, but this could be based on a more generic mechanism, so that it can be reused in other places as well.
Another possibility could be to make use of [wxTimer::SetOwner](https://docs.wxwidgets.org/3.0/classwx_timer.html#aa3966b37329a4fad69ad384733eab27e).
Certainly the current implementation is specially designed for TreeCtrl and thus the receiver of the type wxTreeCtrl is very specific here, but could perhaps also be of the type wxControl or wxEvtHandler. For now, however, the implementation can be left as is, and architectural improvements can be added later.